### PR TITLE
Improve semantic versioning for version integrity

### PR DESCRIPTION
**Pull request by mtojek**
## What does this PR do?

This PR modifies the semantic versioning for version integrity to allow for "-next" entries and prevent duplicated version entries in changelog.

## Checklist

- [x] I have added test packages to [`test/packages`](https://github.com/elastic/package-spec/tree/master/test/packages) that prove my change is effective.
- [x] I have added an entry in [`versions/N/changelog.yml`](https://github.com/elastic/package-spec/blob/master/versions/1/changelog.yml).

## Related issues

- Fixes https://github.com/elastic/package-spec/issues/189

### CODE REVIEW

**Comment by jsoriano** [comment]
Umm, after implementing this, what would be the proper format to include multiple changes in a single version?

**Comment by jsoriano** [comment]
> Umm, after implementing this, what would be the proper format to include multiple changes in a single version?

For example this case looked straightforward to me with two entries with the same version :sweat_smile: https://github.com/elastic/integrations/pull/1153/files#r655491549

**Comment by mtojek** [comment]
Exactly the way as @kaiyan-sheng proposed in https://github.com/elastic/integrations/pull/1153/files#r655601336 :) That's the expected form hence this PR to prevent other options.

File: test/packages/same_version_twice/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- version: "0.0.2"
+  changes:
+    - description: Second
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/193
+- version: "0.0.1"
+  changes:
+    - description: First
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/193
```
Comment (jsoriano) [inline test/packages/same_version_twice/changelog.yml:21]: Is there any test case with a version block with multiple changes?

**Review by jsoriano — APPROVED** [review]
Great! Thanks for the clarifications.

File: test/packages/same_version_twice/changelog.yml (hunk 1 above)
Comment (mtojek) [reply test/packages/same_version_twice/changelog.yml:30]: Good catch! I added multiple entries in few changelogs.

Comment (jsoriano) [reply test/packages/same_version_twice/changelog.yml:30]: Thanks!

**Pull request merged**